config/sections/mixins/layout.php

```diff
@@ -120,6 +123,39 @@
 			}
 
 			return $item;
-		}
+		},
+		'sortColumn' => function () {
+			return App::instance()->request()->get('sortColumn');
+		},
+		'sortData' => function (array $data = []) {
+			$sortColumn    = $this->sortColumn();
+			$sortDirection = $this->sortDirection();
+
+			// check table layout and sort column
+			if ($this->layout !== 'table' || empty($sortColumn) === true) {
+				return $data;
+			}
+
+			usort($data, function ($a, $b) use ($sortColumn, $sortDirection) {
+				if ($sortDirection === 'asc') {
+					if (is_array($a[$sortColumn]) === true) {
+						return $a[$sortColumn]['text'] <=> $b[$sortColumn]['text'];
+					} else {
+						return $a[$sortColumn] <=> $b[$sortColumn];
+					}
+				} else {
+					if (is_array($b[$sortColumn])) {
+						return $b[$sortColumn]['text'] <=> $a[$sortColumn]['text'];
+					} else {
+						return $b[$sortColumn] <=> $a[$sortColumn];
+					}
+				}
+			});
```

Comment:
I really need your help here. I have to use `usort` function because of data is array (multidimensional). I couldn't solve with Kirby way (especially via collections). What do you have better and stable solutions?

Reply:
We could use `array_multisort()` for this, which should be a bit faster than `usort()`.

Untested example:

```suggestion
			// extract the sort column value from each row
			$columnData = array_map(function ($row) use ($sortColumn) {
				$value = $row[$sortColumn];

				if (is_array($value) === true) {
					return $value['text'];
				}

				return $value;
			}, $data);

			// sort $data by the sort column values
			array_multisort(
				$columnData,
				$sortDirection === 'desc' ? SORT_DESC : SORT_ASC,
				$data
			);
```

Reply:
Did you check if A::sort could work here?

Reply:
@bastianallgeier No, `A::sort` doesn't work. Because column value can be array. We need to extract the value like that: `return is_array($value) ? $value['text'] : $value;`

For using `A::sort`, the values in the table layout should be refactored:
https://github.com/getkirby/kirby/blob/develop/config/sections/mixins/layout.php#L98-L103

Current data:
````
Array
(
    [0] => Array
        (
            [title] => Array
                (
                    [text] => Title A
                    [url] => /panel/pages/a
                )
            [info] => Info 1
            [date] => 2020
        )
    [1] => Array
        (
            [title] => Array
                (
                    [text] => Title B
                    [url] => /panel/pages/b
                )
            [info] => Info 2
            [date] => 2021
        )
    [2] => Array
        (
            [title] => Array
                (
                    [text] => Title C
                    [url] => /panel/pages/c
                )
            [info] => Info 3
            [date] => 2022
        )

)
````

We need to normalize values:
````
Array
(
    [0] => Array
        (
            [title] => Title A
            [info] => Info 1
            [date] => 2020
        )
    [1] => Array
        (
            [title] => Title B
            [info] => Info 2
            [date] => 2021
        )
    [2] => Array
        (
            [title] => Title C
            [info] => Info 3
            [date] => 2022
        )

)
````